src/main/java/org/opensha/sha/calc/SpectrumCalculator.java

```diff
@@ -204,7 +205,10 @@ public DiscretizedFunc getIML_SpectrumCurve(DiscretizedFunc spectrumFunction,
 
 		// loop over sources
 		for (sourceIndex = 0; sourceIndex < numSources; sourceIndex++) {
-
+			// quit if user cancelled the calculation
+			if (HazardCurveApplication.isCancelled())
+				return new ArbitrarilyDiscretizedFunc();
+			
```

Comment:
I don't love this static approach here; I would prefer that the application tells the calculator that a cancellation has been requested. Here's what I proposed in my comment on #109:

> The simplest might be to add an AtomicBoolean to HazardCurveCalculator that represents a canceled state with a setter method, and to check if it has been set inside the source loop (probably not the inner rupture loop) of computeHazardCurve and exit if so.

More specifically, add a private `AtomicBoolean` instance to HazardCurveCalculatator that defaults to false and setter method to indicate cancellation (e.g., `setCancelled()`). Then, if you detect cancellation in the loop, reset it to to false and return null (don't return an empty function).